From 36073510926d28de485708510ca5574fc8df0d09 Mon Sep 17 00:00:00 2001
From: Emily Deng <Emily.Deng@amd.com>
Date: Wed, 6 Nov 2019 16:20:54 +0800
Subject: [PATCH 4387/4736] SWDEV-210749 drm/amdgpu: Need to disable msix when
 unloading driver

For driver reload test, it will report "can't enable
MSI (MSI-X already enabled)".

Change-Id: Id98a33e8404d8d803f20d9694f2f04a6e5251fb7
Signed-off-by: Emily Deng <Emily.Deng@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c
index 22edda8ad261..48af4830a74f 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_irq.c
@@ -308,7 +308,11 @@ void amdgpu_irq_fini(struct amdgpu_device *adev)
 		drm_irq_uninstall(adev->ddev);
 		adev->irq.installed = false;
 		if (adev->irq.msi_enabled)
+#ifdef PCI_IRQ_MSI
+			pci_free_irq_vectors(adev->pdev);
+#else
 			pci_disable_msi(adev->pdev);
+#endif
 		if (!amdgpu_device_has_dc_support(adev))
 			flush_work(&adev->hotplug_work);
 	}
-- 
2.17.1

